algorithms/integration/integrator.py

```diff
@@ -1208,6 +1208,19 @@ def integrate(self):
         def _determine_max_memory_needed(experiments, reflections):
             max_needed = 0
             for imageset in experiments.imagesets():
+                # find all experiments belonging to that imageset, as each
+                # imageset is processed as a whole for integration.
+                if all(experiments.identifiers()):
+                    expt_ids = []
+                    for experiment in experiments:
+                        if experiment.imageset == imageset:
+                            expt_ids.append(experiment.identifier)
```

Comment:
```suggestion
                    expt_ids = [
                        experiment.identifier
                        for experiment in experiments:
                        if experiment.imageset == imageset
                    ]
```
just a suggestion: this feels like it should be a list comprehension (or a tuple or even just a generator)